### PR TITLE
feat: add BAL to FE

### PR DESCRIPTION
**Pull request by Tulun**
Quick testing of adding BAL for Korpi and Nick.

### CODE REVIEW

File: src/data/routes_1_0xc186fA914353c44b2E33eBE05f21846F1048bEda.json [hunk 1]
```diff
@@ -336,6 +372,15 @@
...
       "l1TokenAddress": "0x6B175474E89094C44Da98b954EedeAC495271d0F"
     },
+    {
+      "fromChain": 10,
+      "toChain": 137,
+      "fromTokenAddress": "0xFE8B128bA8C78aabC59d4c64cEE7fF28e9379921",
+      "fromSpokeAddress": "0xa420b2d1c0841415A695b81E5B867BCD07Dff8C9",
+      "fromTokenSymbol": "BAL",
+      "isNative": false,
+      "l1TokenAddress": "0x6B175474E89094C44Da98b954EedeAC495271d0F"
```
Comment (kevinuma) [inline src/data/routes_1_0xc186fA914353c44b2E33eBE05f21846F1048bEda.json:382]: This is DAI's address. Should be 0xba100000625a3754423978a60c9317c58a424e3D instead

File: src/utils/config.ts [hunk 2]
```diff
@@ -158,7 +158,9 @@ export class ConfigClient {
   // this has a chance to mix up eth/weth which can be a problem. prefer token by symbol.
   getTokenInfoByAddress(chainId: number, address: string): Token {
     const tokens = this.getTokenList(chainId);
-    const token = tokens.find((token) => token.address === address);
+    const token = tokens.find(
+      (token) => token.address.toLowerCase() === address.toLowerCase()
```
Comment (kevinuma) [inline src/utils/config.ts:162]: Why is this needed now but not before?

File: src/data/routes_1_0xc186fA914353c44b2E33eBE05f21846F1048bEda.json [hunk 3]
```diff
@@ -111,6 +120,15 @@
...
       "l1TokenAddress": "0x6B175474E89094C44Da98b954EedeAC495271d0F"
     },
+    {
+      "fromChain": 1,
+      "toChain": 137,
+      "fromTokenAddress": "0xba100000625a3754423978a60c9317c58a424e3d",
+      "fromSpokeAddress": "0x4D9079Bb4165aeb4084c526a32695dCfd2F77381",
+      "fromTokenSymbol": "BAL",
+      "isNative": false,
+      "l1TokenAddress": "0xba100000625a3754423978a60c9317c58a424e3d"
```
Comment (kevinuma) [inline src/data/routes_1_0xc186fA914353c44b2E33eBE05f21846F1048bEda.json:130]: This is not in the checksummed format. Might be why you needed toLowerCase further below?

File: src/utils/constants.ts [hunk 4]
```diff
@@ -384,6 +384,13 @@ export const tokenList: TokenInfoList = [
     logoURI: "/logos/ethereum-logo.svg",
     mainnetAddress: getAddress("0x7d1afa7b718fb893db30a3abc0cfc608aacfebb0"),
   },
+  {
+    name: "Balancer",
+    symbol: "BAL",
+    decimals: 18,
+    logoURI: "/logos/bal.svg",
+    mainnetAddress: getAddress("0xba100000625a3754423978a60c9317c58a424e3d"),
```
Comment (kevinuma) [inline src/utils/constants.ts:392]: This is also not in checksummed format

**Review by kevinuma — APPROVED** [review]

**Review by amateima — APPROVED** [review]
Will this PR enable BAL deposits? Some changes will be needed on the Scraper API to count referrals, this is why I'm asking

**Review by daywiss — APPROVED** [review]
fyi we have an [automated script](https://github.com/across-protocol/helper-scripts) which creates this routes file based on on-chain settings. i ran this and double checked it against the changes here and it exactly matched, so LGTM!
